### PR TITLE
Update std/lib.rs docs to reflect Rust 2018 usage

### LINKED ISSUE #56544: Adjust reference to unidiomatic syntax in std docs roots
The opening section of the std documentation says this:

```
//! `std` is available to all Rust crates by default, just as if each one
//! contained an `extern crate std;` import at the [crate root]. Therefore the
//! standard library can be accessed in [`use`] statements through the path
//! `std`, as in [`use std::env`], or in expressions through the absolute path
//! `::std`, as in [`::std::env::args`].
```

This information, while accurate in both editions, is unidiomatic in the 2018 edition, in that it refers to extern crate syntax and double colon absolute paths. Perhaps this information could be reworded to be expressed in a more neutral way.

### PR DESCRIPTION
**Pull request by Xaeroxe**
Fixes #56544 

This paragraph was written for Rust 2015.  Since 2018 has been stable for a while I think we can update it.

### CODE REVIEW

**Comment by varkor** [comment]
Rust 2018 doesn't replace Rust 2015: I think it'd be better here to explicitly note the differences between the 2015 and 2018 editions. (There might already be a docs convention for this; I'm not sure.)

**Comment by Xaeroxe** [comment]
People reading the std library for the first time are more likely to be using 2018 since that's the new default for cargo, however I'll defer to established standards if they exist.

**Comment by steveklabnik** [comment]
This is great, thanks! This fixes https://github.com/rust-lang/rust/issues/56544, any chance you could amend the commit message to say

    Fixes #56544 

?

> (There might already be a docs convention for this; I'm not sure.)

We haven't discussed it! Generally, we document things "in the present", so we'd default to Rust 2018. I'd be open to an exception here. @rust-lang/docs ?

**Comment by Xaeroxe** [comment]
Sure!

**Comment by QuietMisdreavus** [comment]
I feel like we should leave in the mention of absolute paths, since they're still a thing in 2015 (unless i'm wrong here?). Otherwise, since 2018 is going to be what most people are writing about in the future, explaining things in terms of that perspective is probably going to be best.

> Generally, we document things "in the present", so we'd default to Rust 2018.

+1 to this.

**Comment by Dylan-DPC** [comment]
ping from triage @QuietMisdreavus waiting for your review on this

File: src/libstd/lib.rs
Comment (QuietMisdreavus) [inline src/libstd/lib.rs]: I don't actually know the full extent of the path changes that were introduced in Rust 2018; did absolute paths like this go away as well?

Comment (Xaeroxe) [reply src/libstd/lib.rs]: They did yeah

**Comment by Xaeroxe** [comment]
Done, thanks for the tip!

**Review by QuietMisdreavus — APPROVED** [review]
I think this is good. Since 2018 is the default now, it doesn't make much sense to include details about `extern crate` statements, and the remaining text also applies to 2015 code, since it can still `use std::env` the same way. Let's get this merged, thanks so much!

**Pull request merged**
